### ISSUE TITLE
Failed to post request to wafv2 with EKS Private Cluster

### ISSUE THREAD

**Issue by Larry-JJ** [kubernetes-sigs/aws-load-balancer-controller#1734]
EKS Version:**1.18**
aws-load-balancer-controller version: **v2_1_0**
I followed the below instructions
https://docs.aws.amazon.com/eks/latest/userguide/alb-ingress.html

The EKS cluster is a private cluster, doesn't have a NAT gateway associated. I have created the below VPC endpoints required by private cluster:
com.amazonaws.<region>.ec2
com.amazonaws.<region>.ecr.api
com.amazonaws.<region>.ecr.dkr
com.amazonaws.<region>.s3 
com.amazonaws.<region>.logs 
com.amazonaws.<region>.sts 
com.amazonaws.<region>.elasticloadbalancing 

I got the below error since there is no wafv2 VPC endpoint. actually, I don't use wafv2 in my ingress.yaml. I think it called internally?

```
# This is the yaml:
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: "ingress3"
  namespace: poc
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/scheme: internal
    alb.ingress.kubernetes.io/target-type: instance
  labels:
    app: ingress3
spec:
  rules:
    - http:
        paths:
          - path: /*
            backend:
              serviceName: "ui-service"
              servicePort: 8384

#This is the error:
Events:
 Type     Reason       Age                   From                                 Message
  ----     ------       ----                  ----                                 -------
  Warning  FailedDeployModel  6m43s    ingress      **Failed** deploy model due to **RequestError**: send request failed caused by: Post "https://wafv2.ap-southeast-1.amazonaws.com/": dial tcp 52.119.184.251:443: i/o timeout
```
**If the worker nodes subnet associates with a NAT gateway, it will be working properly.**

Since I don't integrate with WAF specifically, is there any way to omit this error?

Thank you in advance

**Comment by michaelasper** [comment]
in the helm chart, you can disable the waf/wafv2 plugin

**Comment by kishorj** [comment]
If using helm chart, you can set the following during installation 
`--set enableWaf=false --set enableWaf2=false --set enableShield=false`

If you installed the manifest, you need to pass the following args to the controller 
`--enable-shield=false --enable-waf=false --enable-wafv2=false`

**Comment by Larry-JJ** [comment]
> --enable-shield=false --enable-waf=false --enable-wafv2=false

Thank you @kishorj and @michaelasper, after I set up the below flag, it's working properly.
--enable-shield=false --enable-waf=false --enable-wafv2=false

**Comment by jpriebe** [comment]
We had a similar setup -- EKS nodes in private subnets, no NAT gateway.  We have HTTPS proxies, however.  And we can't disable WAF.  So we worked around this by setting the `https_proxy` and `no_proxy` environment variables.  We had AWS support look at our proposed idea; they never came right out and said "yeah, that's how you do it", but they didn't have any objections, either.  When we tried it, we were able to provision ingress load balancers without all the WAF-related errors.

If you use the helm chart to deploy LBC, you set `env.https_proxy` and `env.no_proxy`.  Be very careful with your `no_proxy` values.  If you don't get it right, the LBC pod will not be seen as healthy.  Be sure you have at least these entries in `no_proxy`:

```
localhost,127.0.0.1,172.20.0.0/16,169.254.169.254,.internal
```

And if you're not super-familiar with how `https_proxy` and `no_proxy` work, I highly recommend reading this:

https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/
